src/pages/Search/SearchFiltersDatePage.tsx

```diff
@@ -1,16 +1,40 @@
+import {format} from 'date-fns';
 import React from 'react';
 import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import DatePicker from '@components/DatePicker';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormOnyxValues} from '@components/Form/types';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import ScreenWrapper from '@components/ScreenWrapper';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
-import Text from '@src/components/Text';
+import Navigation from '@libs/Navigation/Navigation';
+import * as Search from '@userActions/Search';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import INPUT_IDS from '@src/types/form/SearchAdvancedFiltersForm';
+
+const currentDate = format(new Date(), CONST.DATE.FNS_FORMAT_STRING);
 
 function SearchFiltersDatePage() {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+
+    const isDateAfterInitialized = !!(searchAdvancedFiltersForm && searchAdvancedFiltersForm[INPUT_IDS.DATE.AFTER]);
```

Comment:
you have to do a lot of work here to make sure `searchAdvancedFiltersForm` is non null.

This code will be simpler if you do a default value when destructuring array in line 26:
```
const [foo = {}] = useOnyx(...
```

then these 2 checks will be much simpler

Reply:
I extracted default value into separate const and added it just like you wrote, everything works just fine

Reply:
I left a comment here, but given that these filters are all optional, I don't think it makes sense to add a default value. We should just leave them empty and add only what the user selects